misk-sqldelight/src/main/kotlin/misk/sqldelight/RetryingTransacter.kt

```diff
@@ -0,0 +1,107 @@
+package misk.sqldelight
+
+import app.cash.sqldelight.Transacter
+import app.cash.sqldelight.TransactionWithReturn
+import app.cash.sqldelight.TransactionWithoutReturn
+import app.cash.sqldelight.db.OptimisticLockException
+import misk.backoff.ExponentialBackoff
+import wisp.logging.getLogger
+import java.sql.SQLException
+import java.sql.SQLRecoverableException
+import java.sql.SQLTransientException
+import java.time.Duration
+
+private val logger = getLogger<RetryingTransacter>()
+
+// NB: all options should be immutable types as copy() is shallow.
+data class TransacterOptions(
+  val maxAttempts: Int = 3,
+  val minRetryDelayMillis: Long = 100,
+  val maxRetryDelayMillis: Long = 500,
+  val retryJitterMillis: Long = 400,
+)
+
+class RetryingTransacter(
+  private val delegate: Transacter,
```

Comment:
for most apps this isn't going to quite work i think? usually you have to inject like a `@MyAppDb transacter` right

Reply:
I didn't even add an `@Inject` annotation here, the only thing is just calling the constructor for now, I want to think more about how to hook this up with guice - there's some subtleties because, in the test here (for example) the thing the _imlemenents_ `delegate: Transacter` is generated by the gradle plugin.